File: Python/Program Mengecek Kata dan Kalimat Palindrom.py
print("Program Mengecek Kata dan Kalimat Palindrom")
print("visit us hobingoding.com")
print()
# Input Kata
kata = input("Input Kata : ")
temp = ""

# Balik Kata
for i in range(len(kata)-1, -1, -1): 
    temp += kata[i]

# Cek Palindrom
if(kata == temp):
    print("Hasil : Palindrom")
else:
    print("Hasil : Bukan Palindrom")

File: C/Program Mengecek Kata dan Kalimat Palindrom.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main() {
    char kata[50] = {'\0'};
    char temp[50] = {'\0'};
    int n, i, j = 0;

    printf("Program Mengecek Kata dan Kalimat Palindrom\n");
    printf("visit us hobingoding.com\n\n");
    // Input Kata
    printf("Input Kata : ");
    gets(kata);
    n = strlen(kata);

    // Ubah Kata Menjadi Lower Case
    for(i = 0; kata[i]; i++) {
        kata[i] = tolower(kata[i]);
    }

    // Balik Kata
    for(i = n-1; i >= 0; i--) {
        temp[j] = kata[i];
        j++;
    }

    // Cek Palindrom
    if(strcmp(kata, temp) == 0)
        printf("Hasil : Palindrom");
    else
        printf("Hasil : Bukan Palindrom");
    return 0;
}
